vars/spawn.groovy

```diff
@@ -7,14 +7,21 @@ def call(Map args = [:], body = null){
 
     def spec = specForImage(args.image, args.version?: 'latest')
     def checkoutScm = args.checkout_scm ?: true
+    def javaOptions = ''
+    def mavenOptions = ''
+    if (args.image == 'java'){
```

Comment:
Then instead specific options of this just process `args.env` and pass it to pod template. In that way pod `spawn` APi would remain generic.